File: SoapClient.java
import org.oorsprong.www.websamples.countryinfo.CountryInfoService;
import org.oorsprong.www.websamples.countryinfo.CountryInfoServiceSoap;

public class SoapClient {
    public static void main(String[] args) {
        // Instantiate the SOAP service
        CountryInfoService service = new CountryInfoService();
        CountryInfoServiceSoap port = service.getCountryInfoServiceSoap();

        // Call the SOAP method to get the capital city for Kenya
        String capitalCity = port.capitalCity("KE");

        // Print the result
        System.out.println("The capital city of Kenya is: " + capitalCity);
    }
}

File: soap-client.ts
import * as soap from 'soap';

// Define WSDL URL
const wsdlUrl = "http://webservices.oorsprong.org/websamples.countryinfo/CountryInfoService.wso?WSDL";

// Create a SOAP client
soap.createClient(wsdlUrl, (err, client) => {
    if (err) {
        console.error("Error creating SOAP client:", err);
        return;
    }

    // Call the function to get the capital city of Kenya
    client.CapitalCity({ sCountryISOCode: 'KE' }, (err: any, result: any) => {
        if (err) {
            console.error("Error calling SOAP service:", err);
            return;
        }

        console.log(`The capital city of Kenya is: ${result?.CapitalCityResult}`);
    });
});
